### ISSUE TITLE
[Suggestion] feature request...

### ISSUE THREAD

**Issue by snaphappy1** [rodit/SnapMod#60]
Thank you again for this wonderful mod! 

I think it would be cool if you could add a special 'Stealth' button to the snapmod app that automatically toggles all anonymous related features like: anonymous story/snap view, anonymous chat view, hide bitmoji enter/exit from chat, disable screenshot, hide save to camera roll... So you could view stories, snaps, and chats without the other person knowing.

Then when you disable the 'Stealth' button, all the toggles return to their default settings...

Even better, and definitely more difficult to implement: Having the 'Stealth' button on the sc app itself, like on your profile. From there it would toggle all the snapmod settings without having to open snapmod.

It would also be nice if you could add a donation popup in the snapmod app that could remind people to donate every once in a while. Also, since some people don't have crypto, if you could add cashapp or venmo as options since those don't reveal your full name like paypal does. Thanks!

**Comment by Magna-Graecia** [comment]
This would be a nice feature, but only if we still retain the option to enable/disable the features individually as well. I might not necessarily want to toggle them all at once, but I can see the interest in having a single button that can toggle them all for convenience when desired

**Comment by snaphappy1** [comment]
Yes the buttons are mainly for convenience but also because there's not much real estate on the sc profile. Each option could still be toggled individually

Something like this would be really cool: 

![SnapMod_SC_Stealth](https://user-images.githubusercontent.com/104136656/164971589-177bb69a-8531-4ce8-8fea-473aa31d6916.jpg)

_____________________

![SnapMod_Stealth_On](https://user-images.githubusercontent.com/104136656/164971606-c89f76c0-4004-4b86-9415-77a773cec38c.jpg)

_____________________

![SnapMod_Stealth_Off](https://user-images.githubusercontent.com/104136656/164971617-e78feeea-68ce-4834-b2d2-184c6cce73a0.jpg)

_____________________

**Comment by Magna-Graecia** [comment]
> Yes the buttons are mainly for convenience but also because there's not much real estate on the sc profile. Each option could still be toggled individually
> 
> Something like this would be really cool:
> 
> ![SnapMod_SC_Stealth](https://user-images.githubusercontent.com/104136656/164971589-177bb69a-8531-4ce8-8fea-473aa31d6916.jpg)
> 
> ![SnapMod_Stealth_On](https://user-images.githubusercontent.com/104136656/164971606-c89f76c0-4004-4b86-9415-77a773cec38c.jpg)
> 
> ![SnapMod_Stealth_Off](https://user-images.githubusercontent.com/104136656/164971617-e78feeea-68ce-4834-b2d2-184c6cce73a0.jpg)

It looks really good. This would be welcomed of course, as long as each option could still be toggled as stated (any feature that adds convenience is welcomed if it works well and doesn't cause other issues), but my only concern is not overburdening rodit with unnecessary requests. My experience having used snapchat modding tools over the years (used something back in the early days of snap and forgot its name, then went on to use Snapprefs, then SnapFreedom, and now SnapMod) is that keeping up with snapchat updates to make sure your tool still works with the latest version is a chore in itself. Snapchat is known to update frequently and actively try to break these mods. They don't like them, plain and simple, which is why they'll ban when detecting someone using them. The more features your tool has, the more work it is to update it and keep it relevant. 

I can relate to this because I do PC game modding, and when a game releases an update, a lot of my mods tend to break and I need to update them. The more mods I have, the more work it becomes exponentially. If I had to go over and update everything too frequently, I would get burned out if I had too many mods/features to address. The same concept applies here, and I think this is why a lot of snapchat mod tool creators end up just abandoning the project. It might be worth the hassle if you can sell them and make some profit, but from my understanding the reason they don't sell them and simply ask for donations instead is because Snapchat will legally go after those who do actively sell them. I believe Snaptools got some legal threats or a cease and desist from Snap inc because of this at one point. Rodit would of course be better to ask as to why he doesn't sell his tool.

But my point is that if we want to do rodit a favor and give him the biggest thanks we can, it would be to keep our requests at a minimum. Just have the most important core features. This way it's a lot easier to maintain with updates. Adding too many convenience features like this, or a "fit to screen/cropping tool", or even one where someone was requesting the ability to disable the SnapMap just so they don't accidentally swipe into it since they never use it, is adding not only a lot of of extra work in creating it, but most importantly in MAINTAINING it. And for what? So you don't accidentally swipe into the SnapMap tab and now have to lift your finger to press one of the other tabs to get out of it? Is it worth the tradeoff? Not in my opinion. It's all up to rodit though on how much he can manage. Personally I'm happy with the app as it is. I would just like to have "Hide Enter Conversation/Hide Exit Conversation" working because it works well in combination with "Hide Read Receipts," And maybe "Hide Typing Status", but that one isn't that important in my opinion because if you're typing, then you're sending a message, and your friend is going to receive a message notification anyway, right? So what's the difference. Other broken/missing features like "Hide Replay Notifications" or "Stealth Snap Viewing" have easy workarounds.

**Comment by rodit** [comment]
I am actually adding this stealth-mode on a per-chat basis in the next update (and refactoring the entire project to make it more maintainable). Snapchat updates are not at all a concern for me - my other project, dexsearch, makes updating SnapMod take about 1 minute for each Snapchat update (I just run the tool and it finds all the hooks for me with a near 100% success rate).

I agree, I do not plan on implementing many 'convenience' features as they don't really interest me but I enjoy adding cool features that require some challenging reverse engineering to add.

**Comment by Magna-Graecia** [comment]
> I am actually adding this stealth-mode on a per-chat basis in the next update (and refactoring the entire project to make it more maintainable). Snapchat updates are not at all a concern for me - my other project, dexsearch, makes updating SnapMod take about 1 minute for each Snapchat update (I just run the tool and it finds all the hooks for me with a near 100% success rate).
> 
> I agree, I do not plan on implementing many 'convenience' features as they don't really interest me but I enjoy adding cool features that require some challenging reverse engineering to add.

Awesome! I'm very glad to hear this. I wonder why other snapchat mod tool creators decided to abandon their projects. I assumed it was because the updates became too much of a hassle to keep up with and it just wasn't worth it anymore. One day they would just up and disappear. No discontinuation announcement or anything haha. Maybe this was indeed the case because they weren't using something like what you have in dexsearch. Well I'm happy to know that maintaining it won't overburden you. Hopefully we'll have SnapMod for years to come :)

Well then hey man, I challenge you to figure out what's causing "Hide Enter Conversation/Hide Exit Conversation" not to work and fix it haha :P .. If I had that one feature working with everything else being as it is, I'd be completely satisfied with the app for my uses to be honest, but that's just me. The more features the merrier though! As long as it doesn't start to get buggy with conflicts, or a cluttered UI full of useless crap lol
